### PR TITLE
core/config: Reset corrupted config file on start

### PR DESCRIPTION
**Pull request by taratatach**
A corrupted config.json file is not parsable and requiring it throws
  an uncaught error, preventing the app to start.
  Since we couldn't figure out how this file gets corrupted and since it
  can get corrupted by any other software, we catch this error and reset
  the file so the app can start in any case.

Please make sure the following boxes are checked:

- [x] PR is not too big
- [x] it improves UX & DX in some way
- [x] it includes tests matching the implementation changes
- [x] it includes relevant documentation

### CODE REVIEW

File: core/config.js [hunk 1]
```diff
@@ -20,7 +25,17 @@ module.exports = class Config {
       this.reset()
     }
 
-    this.config = require(this.configPath)
+    try {
+      this.config = require(this.configPath)
+    } catch (e) {
+      if (e instanceof SyntaxError) {
+        log.error(`Could not read config file at ${this.configPath}:`, e)
```
Comment (aenario) [inline core/config.js]: Would it makes sense to attempt to remove the broken char before giving up? Could be in a separate PR.

Comment (taratatach) [reply core/config.js]: After discussing it, we decided not to make any attempt at fixing the config file.
:ballot_box_with_check:
